### ISSUE TITLE
missing temporary bam files

### ISSUE THREAD

**Issue by yeroslaviz** [mortazavilab/TALON#108]
When running talon with my three samples i get the following error:

```
$ talon --f Analysis/talon/FAT11673.config --db Analysis/talon/WBcel235_gtf_talon.db --build Cel.WBcel235 --threads 12 --o FAT11673
[ 2022-08-19 11:15:36 ] Started TALON run
[E::hts_open_format] Failed to open file talon_tmp/FAT11673_nanofilt_unsorted.bam
Traceback (most recent call last):
  File "/fs/home/yeroslaviz/.local/bin/talon", line 11, in <module>
    load_entry_point('talon==5.0', 'console_scripts', 'talon')()
  File "/fs/home/yeroslaviz/.local/lib/python3.6/site-packages/talon/talon.py", line 2428, in main
    read_groups, intervals, header_file = procsams.partition_reads(sam_files, datasets)
  File "/fs/home/yeroslaviz/.local/lib/python3.6/site-packages/talon/process_sams.py", line 75, in partition_reads
    n_threads = n_threads)
  File "/fs/home/yeroslaviz/.local/lib/python3.6/site-packages/talon/process_sams.py", line 44, in preprocess_sam
    pysam.sort("-@", str(n_threads), "-o", sorted_bam, sam)
  File "/fs/home/yeroslaviz/.local/lib/python3.6/site-packages/pysam/utils.py", line 75, in __call__
    stderr))
pysam.utils.SamtoolsError: 'samtools returned with error 1: stdout=, stderr=samtools sort: can\'t open "talon_tmp/FAT11673_nanofilt_unsorted.bam": No such file or directory\n'
```

The config file looks like that

```
FAT11673_nanofilt,direct-RNA,Nanopore,Analysis/Nanopore_06.2022/talon/FAT11673_nanofilt_labeled.sam
FAT11673_porechop,direct-RNA,Nanopore,Analysis/Nanopore_06.2022/talon/FAT11673_porechop_labeled.sam
FAT11673_rawData,direct-RNA,Nanopore,Analysis/Nanopore_06.2022/talon/FAT11673_rawData_labeled.sam
```

CanI control the place where temporary files are written? Or how do I debug this problem?

thanks

Assa

**Comment by fairliereese** [comment]
Hi, 

Can you try using the `--tmpDir <your temporary directory>` argument when running TALON? You may have to install the latest commits.

**Comment by fairliereese** [comment]
This problem should be solved by 15790fb96f6eb0d045bfb13be08c90df90feda37.
